### PR TITLE
Chore(Flutter): API Query Predicates Examples

### PR DESCRIPTION
**Pull request by Equartey**
_Issue #, if available:_
https://github.com/aws-amplify/amplify-flutter/issues/1695

_Description of changes:_
Adds examples for using query predicates for Flutter API GraphQL Model helpers. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/graphqlapi/flutter/query-data.mdx [hunk 1]
```diff
@@ -53,3 +53,59 @@ if (firstPageData?.hasNextResult == true) {
...
+Return Posts with a rating less than 5.
+
+```dart
+    const rating = 5;
+    
+    GraphQLRequest<PaginatedResult<Post>> request =
+        ModelQueries.list(Post.classType, where: Post.RATING.lt(rating));
+    final response = await Amplify.API.query(request: request).response;
+
+    List<Post?> data = response.data?.items ?? [];
```
Comment (salihgueler) [inline src/fragments/lib/graphqlapi/flutter/query-data.mdx:110]: I would recommend making the code left aligned like the others.

Also using more and relying on type inference would be nice as well. 

e.g. `data` seems to be a list of <Post?> so we can easily omit type there.

```suggestion
const rating = 5;
    
final request = ModelQueries.list(Post.classType, where: Post.RATING.lt(rating));
final response = await Amplify.API.query(request: request).response;

final data = response.data?.items ?? <Post?>[];
```

**Review by offlineprogrammer — APPROVED** [review]

File: src/fragments/lib/graphqlapi/flutter/query-data.mdx [hunk 2]
```diff
@@ -53,3 +53,59 @@ if (firstPageData?.hasNextResult == true) {
...
+
+Get all Posts by parent ID
+
+```dart
+final blogId = blog.id;
+
+final request =
+    ModelQueries.list(Post.classType, where: Post.BLOG.eq(blogId));
+final response = await Amplify.API.query(request: request).response;
+List<Post?> data = response.data?.items ?? [];
```
Comment (salihgueler) [inline src/fragments/lib/graphqlapi/flutter/query-data.mdx:96]:
```suggestion
final data = response.data?.items ?? <Post?>[];
```

File: src/fragments/lib/graphqlapi/flutter/query-data.mdx [hunk 3]
```diff
@@ -53,3 +53,59 @@ if (firstPageData?.hasNextResult == true) {
...
+Query for equality on a model's attribute.
+
+```dart
+const blogTitle = 'Test Blog 1';
+final queryPredicate = Blog.NAME.eq(blogTitle);
+
+GraphQLRequest<PaginatedResult<Blog>> request =
+    ModelQueries.list<Blog>(Blog.classType, where: queryPredicate);
+final response = await Amplify.API.query(request: request).response;
+Blog? BlogFromResponse = response.data?.items[0];
```
Comment (salihgueler) [inline src/fragments/lib/graphqlapi/flutter/query-data.mdx:83]: final blogFromResponse = response.data?.items.first;

File: src/fragments/lib/graphqlapi/flutter/query-data.mdx [hunk 4]
```diff
@@ -53,3 +53,59 @@ if (firstPageData?.hasNextResult == true) {
...
+### Basic Equality Operator
+
+Query for equality on a model's attribute.
+
+```dart
+const blogTitle = 'Test Blog 1';
+final queryPredicate = Blog.NAME.eq(blogTitle);
+
+GraphQLRequest<PaginatedResult<Blog>> request =
+    ModelQueries.list<Blog>(Blog.classType, where: queryPredicate);
```
Comment (salihgueler) [inline src/fragments/lib/graphqlapi/flutter/query-data.mdx:81]:
```suggestion
final request = ModelQueries.list<Blog>(Blog.classType, where: queryPredicate);
```

**Comment by Equartey** [comment]
> Overall great work 🚀
> 
> I just left some comments to keep the code consistent. Let me know if you have any questions about my comments!

Good suggestions! I've applied all. Thanks.

**Review by salihgueler — APPROVED** [review]

**Review by abdallahshaban557 — APPROVED** [review]
LGTM!

**Pull request merged**